pkg/client/speak/client_stream.go

```diff
@@ -0,0 +1,653 @@
+
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-c.ctx.Done():
+			c.closeWs(false)
+			klog.V(6).Infof("StreamClient.listen() Signal Exit\n")
+			klog.V(6).Infof("StreamClient.listen() LEAVE\n")
+			return
+		case <-ticker.C:
+			ws := c.internalConnect()
+			if ws == nil {
+				klog.V(3).Infof("StreamClient.listen(): Connection is not valid\n")
+				klog.V(6).Infof("StreamClient.listen() LEAVE\n")
+				return
+			}
+
+			// msgType can be binary or text
+			msgType, byMsg, err := ws.ReadMessage()
+			if err != nil {
+				errStr := err.Error()
+				switch {
+				case strings.Contains(errStr, SuccessfulSocketErr):
+					klog.V(3).Infof("Graceful websocket close\n")
+
+					// graceful close
+					c.closeWs(false)
+
+					klog.V(6).Infof("StreamClient.listen() LEAVE\n")
+					return
+				case strings.Contains(errStr, FatalReadSocketErr):
+					klog.V(1).Infof("Fatal socket error: %v\n", err)
+
+					// send error on callback
+					sendErr := c.sendError(err)
+					if sendErr != nil {
+						klog.V(1).Infof("StreamClient.listen(): Fatal socket error. Err: %v\n", sendErr)
+					}
+
+					// fatal close
+					c.closeWs(true)
+
+					klog.V(6).Infof("StreamClient.listen() LEAVE\n")
+					return
+				case strings.Contains(errStr, "Deepgram"):
+					klog.V(1).Infof("StreamClient.listen(): Deepgram error. Err: %v\n", err)
+
+					// send error on callback
+					sendErr := c.sendError(err)
+					if sendErr != nil {
+						klog.V(1).Infof("StreamClient.listen(): Deepgram ErrorMsg. Err: %v\n", sendErr)
+					}
+
+					// close the connection
+					c.closeWs(false)
+
+					klog.V(6).Infof("StreamClient.listen() LEAVE\n")
+					return
+				case (err == io.EOF || err == io.ErrUnexpectedEOF) && !c.retry:
+					klog.V(3).Infof("StreamClient object EOF\n")
+
+					// send error on callback
+					sendErr := c.sendError(err)
+					if sendErr != nil {
+						klog.V(1).Infof("StreamClient.listen(): EOF error. Err: %v\n", sendErr)
+					}
+
+					// close the connection
+					c.closeWs(true)
+
+					klog.V(6).Infof("StreamClient.listen() LEAVE\n")
+					return
+				default:
+					klog.V(1).Infof("StreamClient.listen(): Cannot read websocket message. Err: %v\n", err)
+
+					// send error on callback
+					sendErr := c.sendError(err)
+					if sendErr != nil {
+						klog.V(1).Infof("StreamClient.listen(): EOF error. Err: %v\n", sendErr)
+					}
+
+					// close the connection
+					c.closeWs(true)
+
+					klog.V(6).Infof("StreamClient.listen() LEAVE\n")
+					return
+				}
+			}
+
+			if len(byMsg) == 0 {
+				klog.V(7).Infof("StreamClient.listen(): message empty")
+				continue
+			}
+
+			// inspect the message
+			// if c.cOptions.InspectMessage() {
+			// 	err := c.inspect(byMsg)
+			// 	if err != nil {
+			// 		klog.V(1).Infof("StreamClient.listen(): inspect failed. Err: %v\n", err)
+			// 	}
+			// }
+
+			if c.callback != nil {
+				if msgType == websocket.TextMessage {
```

Comment:
I think if you delete this:
https://github.com/deepgram/deepgram-go-sdk/pull/239/files#diff-ab3fe3fda9954ab4fcd7b7f364aea8f13ba807b4a19daa582a7927244fe67593R400-R402

It will remove this warning.

Reply:
Remove line 400-402
```
				if err != nil {
					klog.V(1).Infof("StreamClient.listen(): router.Message failed. Err: %v\n", err)
				}
```

Reply:
Gotcha!